--- PyQt4_gpl_x11-4.12.3/configure.py.orig	2022-03-14 17:31:08.314472040 +0100
+++ PyQt4_gpl_x11-4.12.3/configure.py	2022-03-14 17:34:16.250120572 +0100
@@ -961,6 +961,7 @@ include(%s)
             else:
                 # Use distutils to get the additional configuration.
                 from distutils.sysconfig import get_config_vars
+                from distutils.sysconfig import get_python_lib
                 ducfg = get_config_vars()
 
                 config_args = ducfg.get("CONFIG_ARGS", "")
@@ -975,18 +976,7 @@ include(%s)
                     dynamic_pylib = "--enable-shared" in config_args
 
                 if dynamic_pylib:
-                    if glob.glob("%s/lib/libpython%d.%d*" % (ducfg["exec_prefix"], py_major, py_minor)):
-                        lib_dir_flag = quote("-L%s/lib" % ducfg["exec_prefix"])
-                    elif 'MULTIARCH' in ducfg and glob.glob('%s/lib/%s/libpython%d.%d*' % (ducfg['exec_prefix'], ducfg['MULTIARCH'], py_major, py_minor)):
-                        lib_dir_flag = quote('-L%s/lib/%s' % (ducfg['exec_prefix'], ducfg['MULTIARCH']))
-                    elif glob.glob("%s/libpython%d.%d*" % (ducfg["LIBDIR"], py_major, py_minor)):
-                        lib_dir_flag = quote("-L%s" % ducfg["LIBDIR"])
-                    else:
-                        sipconfig.inform("Qt Designer plugin disabled because Python library couldn't be found")
-                        lib_dir_flag = ''
-                        opts.designer_plugin = False
-
-                    link = "%s -lpython%d.%d%s" % (lib_dir_flag, py_major, py_minor, abi)
+                    link = "-L%s -lpython%d.%d%s" % (get_python_lib(plat_specific=1, standard_lib=1), py_major, py_minor, abi)
                 else:
                     sipconfig.inform("The Qt Designer plugin was disabled because a dynamic Python library couldn't be found")
                     opts.designer_plugin = False
