diff -urNp -x '*.orig' PyQt4_gpl_x11-4.12.1.org/configure.py PyQt4_gpl_x11-4.12.1/configure.py
--- PyQt4_gpl_x11-4.12.1.org/configure.py	2021-02-28 13:28:36.840739238 +0100
+++ PyQt4_gpl_x11-4.12.1/configure.py	2021-02-28 13:28:37.047403776 +0100
@@ -139,6 +139,8 @@ def create_optparser():
     p.add_option("-g", "--consolidate", action="store_true", default=False,
             dest="bigqt", help="create a single module which links against "
             "all the Qt libraries")
+    p.add_option("--dbus-path", default=None, dest="pydbusmoddir", metavar="DIR",
+		    type="string", help="path to install dbus module")
 
     # These are internal options used to build the mega Windows GPL package.
     p.add_option("--mwg-odbc", action="store_true", default=False,
@@ -1339,7 +1341,10 @@ def check_dbus():
         return
 
     global pydbusmoddir
-    pydbusmoddir = dbus.mainloop.__path__[0]
+    if opts.pydbusmoddir:
+	    pydbusmoddir = opts.pydbusmoddir
+    else:
+	    pydbusmoddir = dbus.mainloop.__path__[0]
 
     # Try and find dbus-python.h.  We don't use pkg-config because it is broken
     # for dbus-python (at least for versions up to and including v0.81.0).
